### PR TITLE
Transfer selected blocks from global_variables to case metadata

### PR DESCRIPTION
**Pull request by perolavsvendsen**
First stab at partial solving of #337 

This PR will transfer selected blocks from `global_variables.yml` into case metadata, when they exist.

### CODE REVIEW

**Comment by perolavsvendsen** [comment]
Note that:
- The list of blocks to fetch from global_variables is predefined in fmu-dataio. E.g. a random user _can not_ specify a "random" block. In other words; a standard is inferred here. Not sure if this actually exists.
- If a requested block does not exist, or has a different type than expected, `warn` and move on. We do not want the generation of case metadata to fail at this point, because debugging that from an end user perspective is close to impossible as long as this runs in a HOOK workflow. (The main FMU run will move on regardless of success of the HOOK or not, so if this fails, user will experience failure on all later uploads but it will not be intuitive what caused it.)

File: schema/definitions/0.8.0/examples/case.yml
[the remark this answers is not in the document]
Comment (perolavsvendsen) [reply schema/definitions/0.8.0/examples/case.yml:1]: In this proposal, only one new block in the (case) metadata, which is `fmu.case.config`. And then "whatever" into that. I.e. we don't validate in detail, so fmu-dataio can technically include anything into that block.

But given the ongoing discussion around `parameters.txt` where we are leaning towards having that as a separate data object, we might discuss this PR in that perspective as well. Should case config be a separate data object? 🤷‍♂️

[the remark this answers is not in the document]
Comment (perolavsvendsen) [reply schema/definitions/0.8.0/examples/case.yml:1]: And, another point - which is possibly what you meant - is access control and information classification. We cannot have a situation where someone is to be granted access to only parts of the case metadata, that would not work.

**Comment by mferrera** [comment]
Closing this PR (as part of a Julevask effort) because the code base has changed so significantly as to be impossible to merge, but the issue remains open.

**Pull request closed without merging**